library/alloc/src/vec/mod.rs

```diff
@@ -1460,6 +1460,21 @@ impl<T, A: Allocator> Vec<T, A> {
     /// # }
     /// ```
     ///
+    /// It is possible to initialize the spare capacity with [`spare_capacity_mut()`]:
+    ///
+    /// ```
+    /// # use core::mem::MaybeUninit;
+    /// let mut vec: Vec<u8> = Vec::with_capacity(1024);
+    /// let remaining = vec.spare_capacity_mut();
+    /// remaining[..512].iter_mut().for_each(|x| *x = MaybeUninit::new(0));
+    /// // SAFETY:
+    /// // 1. The `new_len` is less than the `capacity`.
+    /// // 2. The elements in range `0..new_len` are initialized.
+    /// unsafe {
+    ///     vec.set_len(256);
+    /// }
+    /// ```
```

Comment:
It might even be worth it to move this above the FFI example since this use usable with pure rust.

Reply:
Happy to amend the commit accordingly. Who'd be an authority for such decisions?

Reply:
> Who'd be an authority for such decisions?

Whoever says something that makes the most sense, which could be you :) docs changes are usually EAFP. (agreed that it makes sense to move this up).